src/fu-util.c

```diff
@@ -560,6 +617,227 @@ fu_util_install_prepared (FuUtilPrivate *priv, gchar **values, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_util_history_clear (FuUtilPrivate *priv, gchar **values, GError **error)
+{
+	return fu_pending_remove_all (priv->pending, error);
+}
+
+static gboolean
+fu_util_history_report (FuUtilPrivate *priv, gchar **values, GError **error)
+{
+	guint status_code;
+	g_autofree gchar *config_fn = NULL;
+	g_autofree gchar *data = NULL;
+	g_autofree gchar *machine_hash = NULL;
+	g_autofree gchar *report_uri = NULL;
+	g_autoptr(GKeyFile) kf = g_key_file_new ();
+	g_autoptr(GPtrArray) results = NULL;
+	g_autoptr(JsonBuilder) builder = NULL;
+	g_autoptr(JsonGenerator) json_generator = NULL;
+	g_autoptr(JsonNode) json_root = NULL;
+	g_autoptr(SoupMessage) msg = NULL;
+
+	/* set up networking */
+	if (!fu_util_setup_networking (priv, error))
```

Comment:
I was just looking at d980ccb757ea44fcf24e8f21d676b8a58ee69565 for this since this looked new to me.  I think you also need to set up https_proxy, right?  the request should be going to a SSL URL now.

Reply:
I think that http_proxy is supposed to be used for both, no? If that's not right, then some of my other packages are broken too :)

Reply:
They're separate.  I have to use a proxy on my workstation.
```
$ env | grep proxy
http_proxy=http://127.0.0.1:8123
https_proxy=http://127.0.0.1:8123
$ wget https://fwupd.org
--2018-01-08 15:59:20--  https://fwupd.org/
Connecting to 127.0.0.1:8123... connected.
Proxy request sent, awaiting response... 200 OK
Length: 5205 (5.1K) [text/html]
Saving to: ‘index.html’

index.html                                          100%[==================================================================================================================>]   5.08K  --.-KB/s    in 0s      

2018-01-08 15:59:21 (114 MB/s) - ‘index.html’ saved [5205/5205]

$ unset https_proxy
$ wget https://fwupd.org
--2018-01-08 15:59:27--  https://fwupd.org/
Resolving fwupd.org (fwupd.org)... 51.15.67.4
Connecting to fwupd.org (fwupd.org)|51.15.67.4|:443... 
```

It just hangs for me there.

Reply:
I think it still works for you in fwupdmgr as they are the same value, but I guess something like this would be more safe in case they are different?

    -       http_proxy = g_getenv ("http_proxy");
    +       http_proxy = g_getenv ("https_proxy");
    +       if (http_proxy == NULL)
    +               http_proxy = g_getenv ("http_proxy");

Reply:
Yeah that's probably why it still works in fwupdmgr, yep.
For more people that's probably enough.